File: old/sorting/insertion_sort/insertion_sort_by_string_length/insertion_sort_by_string_length.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;


void insertionSortByStringLength(vector<string>& arr) {
    for (int i = 1; i < arr.size(); i++) {
        string currWord = arr[i];
        int prevWordIndex = i - 1;
        while (prevWordIndex >= 0 && arr[prevWordIndex].size() > currWord.size()) {
            arr[prevWordIndex + 1] = arr[prevWordIndex];
            prevWordIndex -= 1;
        }
        arr[prevWordIndex + 1] = currWord;
    }
}


int main() {
    vector<string> v = {"atodon", "beeeeeeeee", "todo", "coding"};
    insertionSortByStringLength(v);
    for(int i = 0; i < v.size(); i++) {
        cout << v[i] << " ";
    }
    cout << endl;
    return 0;
}

File: old/sorting/insertion_sort/insertion_sort_by_string_length/insertion_sort_by_string_length.py
def insertion_sort_by_string_length(words: list[int]) -> list[int]:
    for i in range(1, len(words)):
        curr_word  = words[i]
        prev_word_index = i - 1
        while (prev_word_index >= 0 and len(words[prev_word_index]) > len(curr_word)):
            words[prev_word_index + 1] = words[prev_word_index]
            prev_word_index -= 1
        words[prev_word_index + 1] = curr_word
    return words
            
def main():
    print(insertion_sort_by_string_length(["code","a","be","go"]))

main()
